assembly/src/assembler/instruction/mem_ops.rs

```diff
@@ -125,7 +125,17 @@ pub fn local_to_absolute_addr(
     }
 
     let max = num_proc_locals - 1;
-    validate_param(index, 0..=max)?;
+    if index > max {
+        return Err(AssemblyError::Other(
+            Report::msg(
+                format!(
+                    "invalid local memory index: {} is out of bounds for procedure with {} locals (valid range: 0..{})",
+                    index, num_proc_locals, max
+                )
+            )
+            .into(),
+        ));
```

Comment:
We actually already have the machinery to output very nice error messages (built on top of `miette`). The desired solution would output an error variant that is [similar to `InvalidSyscallTarget`](https://github.com/0xPolygonMiden/miden-vm/blob/ea39c0587ef20307db807795929a59d6d9a26cba/assembly/src/errors.rs#L55-L63) (in that it takes a `SourceSpan` and a `SourceFile`).